src/main/kotlin/org/opensearch/replication/task/shard/TranslogSequencer.kt

```diff
@@ -66,35 +72,57 @@ class TranslogSequencer(scope: CoroutineScope, private val replicationMetadata:
+                        leaderAlias, leaderIndexName
                     )
-                    if (replayResponse.shardInfo.failed > 0) {
-                        replayResponse.shardInfo.failures.forEachIndexed { i, failure ->
-                            log.error("Failed replaying changes. Failure:$i:$failure}")
+                    replayRequest.parentTask = parentTaskId
+                    launch {
+                        var relativeStartNanos = System.nanoTime()
+                        val retryOnExceptions = ArrayList<Class<*>>()
+                        retryOnExceptions.add(MappingNotAvailableException::class.java)
+                        var tooManyRequestsAcknowledgement = true
+                        while (tooManyRequestsAcknowledgement) {
+                            tooManyRequestsAcknowledgement=false
+                            try {
+                                val replayResponse = client.suspendExecuteWithRetries(
+                                    replicationMetadata,
+                                    ReplayChangesAction.INSTANCE,
+                                    replayRequest,
+                                    log = log,
+                                    retryOn = retryOnExceptions
+                                )
+                                if (replayResponse.shardInfo.failed > 0) {
+                                    replayResponse.shardInfo.failures.forEachIndexed { i, failure ->
+                                        log.error("Failed replaying changes. Failure:$i:$failure}")
+                                    }
+                                    followerClusterStats.stats[followerShardId]!!.opsWriteFailures.addAndGet(
+                                        replayResponse.shardInfo.failed.toLong()
+                                    )
+                                    throw ReplicationException(
+                                        "failed to replay changes",
+                                        replayResponse.shardInfo.failures
+                                    )
+                                }
+                                val tookInNanos = System.nanoTime() - relativeStartNanos
+                                followerClusterStats.stats[followerShardId]!!.totalWriteTime.addAndGet(
+                                    TimeUnit.NANOSECONDS.toMillis(tookInNanos)
+                                )
+                                followerClusterStats.stats[followerShardId]!!.opsWritten.addAndGet(
+                                    replayRequest.changes.size.toLong()
+                                )
+                            } catch (e: OpenSearchException) {
+                                if (e.status() == RestStatus.TOO_MANY_REQUESTS) {
+                                    tooManyRequestsAcknowledgement = true
+                                } else {
+                                    throw e
```

Comment:
Once this exception is thrown, we will need to retry this at the task level.
Could we please check this case as well?